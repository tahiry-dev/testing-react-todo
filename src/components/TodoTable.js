import React from 'react';

function TodoTable({ currentInput, handleInput, handleKeyPress, inputArr, handleCheckboxClick }) {
  return (
    <div className='inputTable'>
      <h3>TODO</h3>
      <input 
        value={currentInput} 
        onChange={handleInput} 
        onKeyDown={handleKeyPress} 
        type="text" 
        placeholder="Enter your todo"
        />
      <ul>
        {inputArr.map((item, index) => (
          <li key={index}><input onClick={() => handleCheckboxClick(index)} type="checkbox"/> {item}</li>
        ))}
      </ul>
    </div>
  );
}

export default TodoTable;