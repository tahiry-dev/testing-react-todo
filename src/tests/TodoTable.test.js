import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import TodoTable from '../components/TodoTable';


describe('TodoTable', () => {

  const currentInput = '';
  const handleInput = jest.fn();
  const handleKeyPress = jest.fn();
  const inputArr = [];
  const handleCheckboxClick = jest.fn();

  it("renders the component correctly", () => {
    const { asFragment } = render(
      <TodoTable 
        currentInput={currentInput} 
        handleInput={handleInput} 
        handleKeyPress={handleKeyPress} 
        inputArr={inputArr} 
        handleCheckboxClick={handleCheckboxClick} 
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the correct input value when it changes', () => {
    const inputArr = ['item1', 'item2'];
    const handleInput = jest.fn();
    const handleKeyPress = jest.fn();
    const handleCheckboxClick = jest.fn();
    const { getByPlaceholderText } = render(
      <TodoTable 
        currentInput="initial value" 
        handleInput={handleInput} 
        handleKeyPress={handleKeyPress} 
        inputArr={inputArr} 
        handleCheckboxClick={handleCheckboxClick}
      />
    );

    const input = getByPlaceholderText('Enter your todo');
    expect(input).toHaveValue('initial value');
  });

  it('should call handleKeyPress when the user presses Enter', () => {
    const handleKeyPress = jest.fn();
    const { getByPlaceholderText } = render(
      <TodoTable 
        currentInput="" 
        handleInput={() => {}} 
        handleKeyPress={handleKeyPress} 
        inputArr={[]} 
        handleCheckboxClick={() => {}} 
      />
    );
    const input = getByPlaceholderText('Enter your todo');
    fireEvent.keyDown(input, { key: 'Enter', code: 13 });
    expect(handleKeyPress).toHaveBeenCalled();
  });


  it('tests onClick event handler', () => {
    const handleCheckboxClick = jest.fn();
    const inputArr = ['Todo 1', 'Todo 2'];
    const { getAllByRole } = render(
      <TodoTable
        inputArr={inputArr}
        handleCheckboxClick={handleCheckboxClick}
      />
    );
    const checkboxes = getAllByRole('checkbox');
    fireEvent.click(checkboxes[0]);
    expect(handleCheckboxClick).toHaveBeenCalledWith(0);
  });

});
