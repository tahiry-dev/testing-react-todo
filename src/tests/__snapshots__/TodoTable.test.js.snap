// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoTable renders the component correctly 1`] = `
<DocumentFragment>
  <div
    class="inputTable"
  >
    <h3>
      TODO
    </h3>
    <input
      placeholder="Enter your todo"
      type="text"
      value=""
    />
    <ul />
  </div>
</DocumentFragment>
`;
